controllers/nodefeaturediscovery_controls.go

```diff
@@ -451,18 +451,21 @@ func Deployment(n NFD) (ResourceStatus, error) {
 	}
 
 	var args []string
-	port := defaultServicePort
 
-	// If the operand service port has already been defined,
-	// then set "port" to the defined port. Otherwise, it is
-	// ok to just use the defaultServicePort value
-	if n.ins.Spec.Operand.ServicePort != 0 {
-		port = n.ins.Spec.Operand.ServicePort
-	}
+	if n.ins.Spec.GrpcMode && obj.Name == nfdMasterApp {
+		port := defaultServicePort
+
+		// If the operand service port has already been defined,
+		// then set "port" to the defined port. Otherwise, it is
+		// ok to just use the defaultServicePort value
+		if n.ins.Spec.Operand.ServicePort != 0 {
+			port = n.ins.Spec.Operand.ServicePort
+		}
 
-	// Now that the port has been determined, append it to
-	// the list of args
-	args = append(args, fmt.Sprintf("--port=%d", port))
+		// Now that the port has been determined, append it to
+		// the list of args
+		args = append(args, fmt.Sprintf("--port=%d", port))
```

Comment:
We should also specify `-enable-nodefeature-api={true|false}` based on GrpcMode, right? And for both nfd-master and nfd-worker.

nit: one dash is enough 😜
```suggestion
		args = append(args, fmt.Sprintf("-port=%d", port))
```